tests/provisioning/provision_test.go

```diff
@@ -111,17 +116,19 @@ func (p *ProvisionTestSuite) TestProvisionDynamicInput() {
 	}
 
 	for _, tt := range tests {
+		tt.name = tt.name + " Module: " + p.terraformConfig.Module + " Kubernetes version: " + p.clusterConfig.KubernetesVersion
+
 		clusterName := namegen.AppendRandomString(provisioning.TFP)
 
 		p.Run((tt.name), func() {
-			provisioning.Provision(p.T(), clusterName, p.terraformConfig, p.clusterConfig, p.terraformOptions)
-			provisioning.VerifyCluster(p.T(), p.client, clusterName, p.terraformConfig, p.terraformOptions, p.clusterConfig)
+			defer cleanup.Cleanup(p.T(), p.terraformOptions)
 
-			cleanup.Cleanup(p.T(), p.terraformOptions)
+			provisioning.Provision(p.T(), tt.client, clusterName, p.clusterConfig, p.terraformOptions)
+			provisioning.VerifyCluster(p.T(), p.client, clusterName, p.terraformConfig, p.terraformOptions, p.clusterConfig)
 		})
 	}
 }
 
-func TestProvisionTestSuite(t *testing.T) {
-	suite.Run(t, new(ProvisionTestSuite))
+func TestTFPProvisionTestSuite(t *testing.T) {
```

Comment:
same nit here for TFPProvision being in the suite name and the test names

Reply:
Resolved